File: sources/julia/14_file_read/main.jl
# task 14 file_read — expected output: 484442112
# build: julia main.jl    run: julia main.jl

function main()
    total = Int64(0)
    open("data.bin", "r") do f
        while true
            chunk = read(f, 1048576)
            isempty(chunk) && break
            s = Int64(0)
            for b in chunk
                s += b
            end
            total += s
        end
    end
    println(total % 4294967296)
end

main()

File: sources/swift/14_file_read/main.swift
// task 14 file_read — expected output: 484442112
// build: swiftc -O -o prog main.swift    run: ./prog

import Foundation

let chunkSize = 1 << 20

guard let handle = FileHandle(forReadingAtPath: "data.bin") else {
    fatalError("cannot open data.bin")
}

var total: UInt64 = 0

// One mebibyte per read; a read of nil or of zero bytes means end of file.
while true {
    guard let chunk = try? handle.read(upToCount: chunkSize), !chunk.isEmpty else {
        break
    }
    for byte in chunk {
        total += UInt64(byte)
    }
}

handle.closeFile()

print(total % 4294967296)
